training/MESSAGING_OVERVIEW.md

```diff
@@ -0,0 +1,138 @@
+var arg1 = msgArgs.getValueOf("arg1");
+var arg2 = msgArgs.getValueOf("arg2");
+```
+
+Now the values of `arg1` and `arg2` are available for use within this Chapel function.
+
+<a id="newModule"></a>
+## Creating a New Chapel Module
+
+Adding a new Chapel module is simple. Create a file of the same name as your Chapel module. If there is no communication between Client and Server in your module, there's nothing special needed! Otherwise, there are two important things to remember when creating a new module that have communication between Client and Server:
+
+- Add the module name to `serverModules.cfg`. (This file sits at the root level of the Arkouda repo).
+- At the end of your Chapel module, include `use CommandMap;` followed by the `registerFunction(stringName, functionName, moduleName)` method for each function in your module that will receive a request from the Client.
+
+Putting everything together, the client's request to the server will look something like this:
+
+```
+arg1 = "First Argument"
+arg2 = "Second Argument"
+repMsg = cast(str, generic_msg(cmd="serverFunction", args={"arg1": arg1, "arg2": arg2}))
+```
+
+<a id="exMod"></a>
+Using this request, this is an example of a Chapel module that will process and return a string based on the given arguments `arg1` and `arg2`:
+```chapel
+module ExampleModule {
+
+    use ServerConfig;
+    
+    use Reflection;
+    use ServerErrors;
+    use Message;
+
+    use MultiTypeSymbolTable;
+    use MultiTypeSymEntry;
+    use ServerErrorStrings;
+    
+    proc exampleFunction(cmd: string, payload: string, argSize: int, 
+                                    st: borrowed SymTab): MsgTuple throws {
+        var repMsg: string; // response message
+
+        var msgArgs = parseMessageArgs(payload, argSize);
+        var arg1 = msgArgs.getValueOf("arg1");
+        var arg2 = msgArgs.getValueOf("arg2");
+        
+        repMsg = "arg1: %s + arg2: %s".format(arg1, arg2);
+        
+        return new MsgTuple(repMsg, MsgType.NORMAL);
+    }
+    
+    use CommandMap;
+    registerFunction("serverFunction", exampleFunction, getModuleName());
+}
+```
+
+If we were to print the value of `repMsg` on the client after performing this request, the output would be: 
+
+`'arg1: First Argument + arg2: Second Argument'`
+
+<a id="tryIt"></a>
+# Try It!
+
+Now it's your turn to use the information presented in this document to create your own module complete with a request to the Server and response back to the Client!
+
+1. In the root-level directory `arkouda`, create a new python file "mymodule.py"
+   - At a minimum for server communication, you will need to import:
+     - `from typing import cast`
+     - `from arkouda.client import generic_msg`
+2. Create a new function which takes in as many arguments as you want
+   - `def my_request(arg1, arg2, ...):`
+3. Send your arguments to the Server
+   - `repMsg = cast(str, generic_msg(cmd="myServerFunction", args={"arg1": arg1, "arg2": arg2}))`

```

Comment:
since `my_request` has arguments `arg1, arg2, ...`, I feel like so should `args`

Reply:
Added ellipsis to `args`